### ISSUE TITLE
What is the expected visibility of associationIdKeys and associationKeys

### ISSUE THREAD

**Issue by jackson-dean** [miragejs/miragejs#298]
# Bug or question

I have a custom serializer which makes use of associationIdKeys and associationKeys to map the id attribute found in `attrs` back to the name of the key which is used to access the associated model. This was broken when these properties were changed to Set instead of Array instances (https://github.com/miragejs/miragejs/pull/151), which is fine, but it makes me wonder if our custom serializer should not be using those fields (they are not prefixed with underscore, but they are also not explicitly documented). Is there a better way to achieve what we are doing. For a very basic example of the type of thing we were doing, consider this snippet:

```js
const customSerializedResult = Object.keys(modelInstance.attrs).reduce((result, currKey) => {
  if (modelInstance.associationIdKeys.includes(currKey)) {
    // ... get associationKey and retrieve nested model to serialize it
  } 
  
  // ... handle top level property
});
```

Is there a different or better way to be doing this?

**Comment by jackson-dean** [comment]
@samselikoff After scanning through the serializer that ships with mirage, it looks it just makes the assumption that things suffixed with `Id` are foreign keys. I thought about doing something like that as well, but felt it might be assuming too much (that no models would have a field ending with `Id`, etc). Anyway, would appreciate any thoughts you have about this :)

**Comment by samselikoff** [comment]
Yes I haven't forgotten about this issue, sorry about the delay. Just need to carve out a bit of time to respond.

**Comment by samselikoff** [comment]
Working on this now should have something for you soon.

Also just re-reading your code you can call `server.config(options)` as many times as you want, so instead of doing

```js
this.server.schema.registerModel(
  'User',
  Model.extend({
    account: belongsTo('Account'),
  })
);
this.server.schema.registerModel('Account', Model);
```

which uses `schema.registerModel` which is private and more likely to break you can do

```js
this.server.config({
  models: {
    user: Model.extend({
      account: belongsTo('Account'),
    }),
    account: Model
  })
);
```

(realizing now server.config isn't documented anywhere but it is public!)

**Comment by samselikoff** [comment]
@jackson-dean believe https://github.com/miragejs/miragejs/pull/325 should fix this, could you take a quick look at this and let me know if it meets your needs?

New APIs are

```js
model.associations
schema.associationsFor('user')
```

Either of these will return an object of all associations, which you can then use to get the `name`, `modelName`, `type` and `foreignKey` from. I think that should get you everything you need?

**Comment by samselikoff** [comment]
Released in v0.1.35, let me know if it works for you!

https://github.com/miragejs/miragejs/releases/tag/v0.1.35

**Comment by jackson-dean** [comment]
@samselikoff Haven't had a chance to upgrade and check it yet, but it looks like it should work fine. Thanks so much!!
